executor/new_executor.go

```diff
@@ -1021,3 +972,38 @@ func (e *MaxOneRowExec) Next() (*Row, error) {
 	}
 	return nil, nil
 }
+
+// TrimExec truncates src rows.
```

Comment:
TrimExec removes hidden fields from src rows. Those hidden fields are used by having/orderby clause but not in select field list.